packages/ec-core/src/test_results.rs

```diff
@@ -349,4 +338,59 @@ mod test_results_from_vec {
         assert_eq!(test_results.results, results.collect::<Vec<_>>());
         assert_eq!(test_results.total_result, scores.into_iter().sum());
     }
+
+    #[test]
+    fn can_clone_test_results_of_scores_of_i32() {
+        let scores = vec![5, 8, 0, 9];
+        let test_results: TestResults<Error<i32>> = scores.into();
+        #[expect(clippy::redundant_clone, reason = "We're testing Clone")]
+        let cloned_test_results = test_results.clone();
+        assert_eq!(cloned_test_results, test_results);
+    }
+
+    #[test]
+    fn can_clone_test_results_of_errors_of_i32() {
+        let errors = vec![5, 8, 0, 9];
+        let test_results: TestResults<Error<i32>> = errors.into();
+        #[expect(clippy::redundant_clone, reason = "We're testing Clone")]
+        let cloned_test_results = test_results.clone();
+        assert_eq!(cloned_test_results, test_results);
+    }
```

Comment:
does this really provide any value? we could just do

```rs
static_assertions::assert_impl_all!(TestResults<Error<i32>>: Clone);
```

instead if you insist but this unit test doesn't make much sense to me since we aren't actually testing any functionality/code we wrote and as such this more checks that the rust compilers derive is working correctly - which is a thing we could do I guess but I'd say we/rust has bigger issues if that breaks.

Instead we are really testing that this compiles, which the `assert_impl_all` could do just as well.

Reply:
But honestly I think that isn't even necessary since removing a derive is pretty hard to do by accident, and I don't see any other reason for why this test could fail.

Reply:
I agree with all this. I think I was focused on "how do I test that I added `.clone()` correctly and went overboard on the tests.

In some sense I was trying to have the tests say "These should be cloneable", in part because removing `Clone` doesn't (I think) break anything inside `ec-core`. So maybe we do want the `assert_impl` just as a way of saying explicitly that we _need_ `Clone` and aren't just including it "out of habit".